pkgs/development/libraries/libtomcrypt/default.nix

```diff
@@ -21,20 +21,32 @@ stdenv.mkDerivation rec {
     })
   ];
 
-  nativeBuildInputs = [ libtool libtommath ];
+  nativeBuildInputs = [ libtool ];
+  buildInputs = [ libtommath ];
+
+  outputs = [ "out" "dev" ];
 
   postPatch = ''
     substituteInPlace makefile.shared --replace "LIBTOOL:=glibtool" "LIBTOOL:=libtool"
   '';
 
+  # Flags needed only when building libtomcrypt
+  env.BUILD_CFLAGS = "-DUSE_LTM";
+  # Flags needed by libtomcrypt and dependent packages
+  env.USER_CFLAGS = "-DLTM_DESC -DLTC_PTHREAD";
+
   preBuild = ''
     makeFlagsArray+=(PREFIX=$out \
-      CFLAGS="-DUSE_LTM -DLTM_DESC -DLTC_PTHREAD" \
+      CFLAGS="$BUILD_CFLAGS $USER_CFLAGS" \
```

Comment:
Nit: This particular flag would probably be better solved using `makeFlags` and the others likely too.